tests/unit/logic/vote.spec.js

```diff
@@ -0,0 +1,1012 @@
+					vote.undo(trs, block, sender, callback)
+				})
+
+				it('should not call to scope.account.merge()', function () {
+					clock.runAll()
+					var result = Vote.__get__('self.scope')
+					expect(result.account.merge.called).to.be.false
+					expect(callback.called).to.be.true
+				})
+			})
+
+			context('if trs.asset.votes is empty', function () {
+
+				beforeEach(function () {
+					trs = {asset: {votes: []}}
+					vote.undo(trs, block, sender, callback)
+				})
+
+				it('should not call to scope.account.merge()', function () {
+					clock.runAll()
+					var result = Vote.__get__('self.scope')
+					expect(result.account.merge.called).to.be.false
+					expect(callback.called).to.be.true
+				})
+			})
+
+			context('if trs.asset.votes is undefined', function () {
+
+				beforeEach(function () {
+					trs = {asset: {votes: undefined}}
+					vote.undo(trs, block, sender, callback)
+				})
+
+				it('should not call to scope.account.merge()', function () {
+					clock.runAll()
+					var result = Vote.__get__('self.scope')
+					expect(result.account.merge.called).to.be.false
+					expect(callback.called).to.be.true
+				})
+			})
+
+			context('if trs.asset.votes has not a valid format', function () {
+
+				beforeEach(function () {
+					trs = {asset: {votes: [1,2]}}
+					vote.undo(trs, block, sender, callback)
+				})
+
+				it('should not call to scope.account.merge()', function () {
+					clock.runAll()
+					var result = Vote.__get__('self.scope')
+					expect(result.account.merge.called).to.be.false
+					expect(callback.called).to.be.true
+				})
+			})
+
+			context('if trs.asset.votes is an Array and have items', function () {
+
+				beforeEach(function () {
+					trs = {asset: {votes: ['+37b6d3fc89fb418fa5e3799e760cc2b1733a567daebb08a531d3d7e8b24f2d22', '+b94d27b9934d3e08a52e52d7da7dabfac484efe37a5380ee9088f7ace2efcde9']}}
+					vote.undo(trs, block, sender, callback)
+				})
+
+				it('should call to scope.account.merge() and callback()', function () {
+					clock.runAll()
+					var result = Vote.__get__('self.scope')
+					expect(result.account.merge.called).to.be.true
+					expect(callback.called).to.be.true
+				})
+			})
+		})
+
+		describe('applyUnconfirmed', function () {
+			var clock, trs, sender, callback, scope
+
+			beforeEach(function () {
+				callback = sinon.spy()
+				clock = sinon.useFakeTimers()
+				sender = {address: 123}
+				scope = { account: { merge: function (address, data, cb) { cb() } } }
+				trs = {asset: {votes: []}}
+				Vote.__set__('setImmediate', setImmediate)
+				Vote.__set__('self.checkUnconfirmedDelegates', function (trs, cb) {
+					return setImmediate(cb)
+				})
+				Vote.__set__('self.scope', scope)
+				vote.applyUnconfirmed(trs, sender, callback)
+			})
+
+			afterEach(function () {
+				clock.restore()
+			})
+
+			it('should call to callback', function () {
+
+				clock.runAll()
+				expect(callback.called).to.be.true
+			})
+		})
+
+		describe('undoUnconfirmed()', function () {
```

Comment:
All this tests are failing :(

Reply:
The reason is because that this test depends on solve the next issue:

https://github.com/RiseVision/rise-node/issues/30

When that issue is solved the test will works properly.